### DISCUSSION TITLE
ImGUI image load helpers

### DISCUSSION THREAD

**Discussion by KingRikkie** [dotnet/Silk.NET#900] [Q&A]
I am struggling to find any example on how to load an image from file with Silk.NET. Using Veldrid this is as simple as: 
```cs
var img = new ImageSharpTexture("image.png");
var dimg = img.CreateDeviceTexture(_gd, _gd.ResourceFactory);
ImgPtr = _controller.GetOrCreateImGuiBinding(_gd.ResourceFactory, dimg); // this returns the ptr to pass to ImGUI.Image()
[. . .]
ImGUI.Image(ImgPtr, new Vector2(300, 300));
```

**Comment by Perksey** [comment]
I don't think our OOTB ImGui controller supports this. We'd welcome contributions in adding something similar to GetOrCreateImGuiBinding for our controller. Hope this helps.

**Reply by KingRikkie** [reply]
Oh that is very unfortunate, I will have to switch libraries then as I lack the deeper knowledge of implementing such a controller myself.

**Comment by KingRikkie** [comment]
Ok so I figured it out myself, modify [Texture.cs](https://github.com/dotnet/Silk.NET/blob/main/examples/CSharp/OpenGL%20Tutorials/Tutorial%201.4%20-%20Textures/Texture.cs#L10) and make the private `_handle` visible from outside. Then in OnLoad:  
```cs
imgTex = new Texture("myTex.png");
imgTex.Bind(TextureUnit.Texture0);
```
And finally in OnRender:
```cs
ImGuiNET.ImGui.Image(new IntPtr(imgTex._handle), new Vector2(300, 300));
```
Don't really understand why ppl recommended a full custom controller instead.
